golog.go

```diff
@@ -206,25 +229,60 @@ type Logger interface {
 	AsStdLogger() *log.Logger
 }
 
+// shouldEnableTrace returns true if tracing was enforced through a linker
+// flag, or if TRACE=true is set in the environment, while favoring the former.
+//
+// See the top-level comment for more information.
+func shouldEnableTrace(currentPrefix string) bool {
+	// Linker flag checks
+	// ------------------
+	if strings.ToLower(linkerFlagEnableTraceThroughLinker) == "true" ||
+		// if trace through linker flags is set, check if the current prefix is
+		// included in the list of prefixes to trace
+		linkerFlagEnableTraceThroughLinker == "1" {
+		// Only return true if the current prefix is included in the list of
+		// prefixes to trace
+		prefixesToTrace := strings.Split(linkerFlagTracePrefixes, ",")
+		for _, prefix := range prefixesToTrace {
+			if strings.ToLower(prefix) == strings.ToLower(currentPrefix) {
+				return true
+			}
+		}
+		return true
+	}
+
+	// Environment variable checks
+	// ---------------------
+	// If TRACE=true is set in the environment, return true
+	envVar := os.Getenv("TRACE")
+	if envVar == "" {
+		return false
+	}
+	if v, err := strconv.ParseBool(os.Getenv("TRACE")); err == nil && v {
+		return true
+	}
+	// Else, this could be a comma-separated list of prefixes to trace
+	// If the current prefix is included in the list, return true
+	for _, prefix := range strings.Split(envVar, ",") {
```

Comment:
Nit - This is a bit of a DRY violation. You could make a function `shouldEnableTrace(currentPrefix, traceConfig string)` and call it twice, once with the linker flag and once with the environment variable.

Reply:
Fully hear you, but I'd hate to have a `traceConfig` var that has `linker` or `env` options.